platforms/android/tangram/src/main/java/com/mapzen/tangram/Marker.java

```diff
@@ -219,16 +219,18 @@ private boolean setBitmap(@NonNull final Bitmap bitmap) {
         bitmap.getPixels(argb, 0, width, 0, 0, width, height);
 
         final int[] abgr = new int[width * height];
-        int row, col;
+        int row = 0, col = 0;
         for (int i = 0; i < argb.length; i++) {
-            col = i % width;
-            row = i / width;
             final int pix = argb[i];
```

Comment:
You can try to use `bitmap.getPixel` in order to remove initial alloc on line `218`, and maybe `bitmap.flip(1f, -1f, pixel_x, pixel_y)` to remove index flipping.

Reply:
Hmm, there is no mention of `bitmap.flip` in the [Android documentation for Bitmap](https://developer.android.com/reference/android/graphics/Bitmap).

Reply:
Can you elaborate? Not sure what these methods are

Reply:
I thought `flip` was a thing but it's not you're right @westnordost , for the first one it's https://developer.android.com/reference/android/graphics/Bitmap.html#getPixel(int,%20int).

Reply:
I have tested this change on around 180 markers and drawing them with this change took around 35 seconds and original code took around 30 seconds.

Reply:
I claim there is something odd with your test setup. I haven't seen any CPU architecture where an integer-division is faster than doing an increment.

Did you execute both in debug mode? Is the project on Github?

Reply:
I wrote a test myself now. The new code is about 16-22% faster.

See measuring code: https://gist.github.com/westnordost/dc4d8cf93a70c0a68853f7c2a545a8a4

Much less than I expected, to be honest. So even with this merged, the performance issue observed in #1896 would still be visible somewhat. I think @matteblair 's idea to either move this "flipping"+"convert argb to abgr" to the native code or feed OpenGL the argb directly (if I understood correctly) is better. However, I am not deep enough in this project to do this, I can only offer this PR. Even if you plan to move it to native code some time later, it still can't hurt to merge this here, for the meantime and since it is already a small improvement.

Reply:
I am sorry, I have done something wrong in the test. Unfortunately I cannot show actual code but basically it is showing markers 180 on map and setting drawable to marker by calling `Marker.setDrawable()` which is calling `Marker.SetBitmap()`. I have rerun test several times and now I am getting same results for both versions - 30 seconds. So I think that in real-world use case this change has negligible performance advantage. Maybe it has something to do with Android JVM optimizations for repeated instructions, JIT, caching. I am not sure, but I have these results.